File: CS/MyCustomLayeredWindow.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using DevExpress.Utils.Internal;
using DevExpress.XtraBars;

namespace WindowsFormsApplication516 {
    class MyDXLayeredWindow : DXLayeredWindowEx {
        Form parent;
        Control ownerControl;
        Image image;
        public MyDXLayeredWindow(Control ownerControl, Image image) {
            this.image = image;
            parent = ownerControl.FindForm();
            this.ownerControl = ownerControl;
            base.Create(parent);
            this.Size = this.image.Size;
            parent.LocationChanged += parent_LocationChanged;
        }

        void parent_LocationChanged(object sender, EventArgs e) {
            Show();
        }
        public void Show() {
            Point p = parent.PointToScreen(ownerControl.Location);
            p.Offset(ownerControl.Width - DrawRect.Width / 2, DrawRect.Height / -2);
            base.Show(p);
            base.Update();
        }
        protected Rectangle DrawRect {
            get {
                Rectangle r = new Rectangle(Point.Empty, Size);
                return r;
            }
        }
        protected override IntPtr hWndInsertAfter {
            get {
                return new IntPtr(0);
            }
        }
        protected override void DrawCore(DevExpress.Utils.Drawing.GraphicsCache cache) {
            cache.Graphics.Clear(Color.Transparent);
            cache.Graphics.DrawImage(image, DrawRect);
        }
    }
    class MyBarDXLayeredWindow : DXLayeredWindowEx {
        Form parent;
        BarItem owner;
        Image image;
        public MyBarDXLayeredWindow(BarItem owner, Image image) {
            this.image = image;
            parent = owner.Manager.Form.FindForm();
            this.owner = owner;
            base.Create(parent);
            this.Size = this.image.Size;
            parent.LocationChanged += parent_LocationChanged;
        }

        void parent_LocationChanged(object sender, EventArgs e) {
            Show();
        }
        public void Show() {
            Point p = parent.PointToScreen(parent.PointToClient(owner.Links[0].ScreenBounds.Location));
            p.Offset(owner.Links[0].Bounds.Width - (DrawRect.Width / 2), DrawRect.Height / -2);
            base.Show(p);
            base.Update();
        }
        protected override IntPtr hWndInsertAfter {
            get {
                return new IntPtr(0);
            }
        }
        protected Rectangle DrawRect {
            get {
                Rectangle r = new Rectangle(Point.Empty, Size);
                return r;
            }
        }
        protected override void DrawCore(DevExpress.Utils.Drawing.GraphicsCache cache) {
            cache.Graphics.Clear(Color.Transparent);
            cache.Graphics.DrawImage(image, DrawRect);
        }
    }
}

File: VB/MyCustomLayeredWindow.vb
Imports System
Imports System.Collections.Generic
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms
Imports DevExpress.Utils.Internal
Imports DevExpress.XtraBars

Namespace WindowsFormsApplication516
    Friend Class MyDXLayeredWindow
        Inherits DXLayeredWindowEx

        Private parent As Form
        Private ownerControl As Control
        Private image As Image
        Public Sub New(ByVal ownerControl As Control, ByVal image As Image)
            Me.image = image
            parent = ownerControl.FindForm()
            Me.ownerControl = ownerControl
            MyBase.Create(parent)
            Me.Size = Me.image.Size
            AddHandler parent.LocationChanged, AddressOf parent_LocationChanged
        End Sub

        Private Sub parent_LocationChanged(ByVal sender As Object, ByVal e As EventArgs)
            Show()
        End Sub
        Public Sub Show()
            Dim p As Point = parent.PointToScreen(ownerControl.Location)
            p.Offset(ownerControl.Width - DrawRect.Width \ 2, DrawRect.Height \ -2)
            MyBase.Show(p)
            MyBase.Update()
        End Sub
        Protected ReadOnly Property DrawRect() As Rectangle
            Get
                Dim r As New Rectangle(Point.Empty, Size)
                Return r
            End Get
        End Property
        Protected Overrides ReadOnly Property hWndInsertAfter() As IntPtr
            Get
                Return New IntPtr(0)
            End Get
        End Property
        Protected Overrides Sub DrawCore(ByVal cache As DevExpress.Utils.Drawing.GraphicsCache)
            cache.Graphics.Clear(Color.Transparent)
            cache.Graphics.DrawImage(image, DrawRect)
        End Sub
    End Class
    Friend Class MyBarDXLayeredWindow
        Inherits DXLayeredWindowEx

        Private parent As Form
        Private owner As BarItem
        Private image As Image
        Public Sub New(ByVal owner As BarItem, ByVal image As Image)
            Me.image = image
            parent = owner.Manager.Form.FindForm()
            Me.owner = owner
            MyBase.Create(parent)
            Me.Size = Me.image.Size
            AddHandler parent.LocationChanged, AddressOf parent_LocationChanged
        End Sub

        Private Sub parent_LocationChanged(ByVal sender As Object, ByVal e As EventArgs)
            Show()
        End Sub
        Public Sub Show()
            Dim p As Point = parent.PointToScreen(parent.PointToClient(owner.Links(0).ScreenBounds.Location))
            p.Offset(owner.Links(0).Bounds.Width - (DrawRect.Width \ 2), DrawRect.Height \ -2)
            MyBase.Show(p)
            MyBase.Update()
        End Sub
        Protected Overrides ReadOnly Property hWndInsertAfter() As IntPtr
            Get
                Return New IntPtr(0)
            End Get
        End Property
        Protected ReadOnly Property DrawRect() As Rectangle
            Get
                Dim r As New Rectangle(Point.Empty, Size)
                Return r
            End Get
        End Property
        Protected Overrides Sub DrawCore(ByVal cache As DevExpress.Utils.Drawing.GraphicsCache)
            cache.Graphics.Clear(Color.Transparent)
            cache.Graphics.DrawImage(image, DrawRect)
        End Sub
    End Class
End Namespace
